src/celengine/astrodb.cpp

```diff
@@ -339,3 +339,18 @@ AstroCatalog::IndexNumber AstroDatabase::getAutoIndex()
     }
     return AstroCatalog::InvalidIndex;
 }
+
+float AstroDatabase::avgDsoMag() const
+{
+    float avg = 0;
+    size_t n = m_dsos.size();
+    for(const auto & dso : m_dsos)
+    {
+        if (dso->getAbsoluteMagnitude() > 8)
+            avg += dso->getAbsoluteMagnitude();
+        else
+            n--;
```

Comment:
inverted logic :) why not just `n=0; ...avg +=...;n++;`?

Reply:
Such an approach was in original code. So, why not...